python-package/lightgbm/sklearn.py

```diff
@@ -746,9 +746,13 @@ def fit(
 
         # Do not modify original args in fit function
         # Refer to https://github.com/microsoft/LightGBM/pull/2619
-        eval_metric_list = copy.deepcopy(eval_metric)
-        if not isinstance(eval_metric_list, list):
-            eval_metric_list = [eval_metric_list]
+        eval_metric_list: List[Union[str, _LGBM_ScikitCustomEvalFunction]]
+        if eval_metric is None:
+            eval_metric_list = []
+        elif isinstance(eval_metric, list):
+            eval_metric_list = copy.deepcopy(eval_metric)
+        else:
+            eval_metric_list = [copy.deepcopy(eval_metric)]
```

Comment:
This refactoring adds an explicit case for `eval_metric is None`. `mypy` can't tell, I guess, that `isinstance(x, list)` would must mean something isn't `None`.